15.md

```diff
@@ -243,6 +247,110 @@ Create a customized user experience using the `naddr` from [NIP-19](https://gith
 
 This event leverages naddr to enable comprehensive customization and sharing of marketplace configurations, fostering a unique and engaging marketplace environment.
 
+## Auctions
+
+### Event `30020`: Create or update a product sold as an auction
+
+**Event Content**:
+```json
+{
+    "id": <String, UUID generated by the merchant. Sequential IDs (`0`, `1`, `2`...) are discouraged>,
+    "stall_id": <String, UUID of the stall to which this product belong to>,
+    "name": <String, product name>,
+    "description": <String (optional), product description>,
+    "images": <[String], array of image URLs, optional>,
+    "starting_bid": <int, denominated in SATS>,
+    "start_date": <int UNIX timestamp, date the auction started / will start, or null>,
+    "duration": <int, number of seconds the auction will run for, excluding eventual time extensions that might happen>,
+    "specs": [
+        [<String, spec key>, <String, spec value>]
+    ],
+    "shipping": [
+        {
+            "id": <String, UUID of the shipping zone. Must match one of the zones defined for the stall>,
+            "cost": <float, extra cost for shipping. The currency is defined at the stall level>,
+        }
+    ]
+}
+```
+
+NB: items sold as an auction are very similar in structure to fixed-price items, with some important differences worth noting.
+
+* While fixed-price items can be priced in any currency and the conversion to Bitcoin happens on checkout, bids on items sold as an auction happen in sats. Client implementations will probably display conversions to various fiat currencies, but the bidder always commits to an amount to be paid in sats if they will happen to be the top bidder at the end of the auction.
```

Comment:
I think other currencies *might* work, but it would be pretty strange, because the payment is always in sats.

For fixed-price items it is good, because "what you see is what you get" - you see 50$ and the amount of sats, you click "buy" and you pay that amount of sats.

For auctions it is strange. Say I commit to buying this item for 50$ (= x sats) (remember, a bid is a commitment!) - but one week later, when I need to actually pay, I realize that I don't have enough sats in my wallet. So I have no way of paying what I committed to pay!

I am happy to not have this limitation in the NIP though - and if other implementations choose to allow other currencies, they would be free to do so.

How would you do that? Specify a `bid_currency` on the auction object and then all amounts would be in that currency? Sounds simple enough and worth it for keeping the NIP as general as possible!

Reply:
> How would you do that? Specify a bid_currency on the auction object

The `auction object` is a particular type of product. This means it belongs to a `Stall` and the `Stall` has a `currency` that can be used.

Reply:
Interesting, but does that mean I need a separate stall if I want to run auctions? I can't have auctions and fixed-price items on the same stall?

Reply:
> I can't have auctions and fixed-price items on the same stall?
- yes, it should possible